### PR TITLE
Centralize template logic

### PR DESCRIPTION
**Pull request by signed8bit**
In preparation for the changes needed to address #463, I wanted to propose a slight cleanup of the duplicate cookiecutter code.

### CODE REVIEW

File: test/unit/test_util.py [hunk 1]
```diff
@@ -135,8 +137,34 @@ def test_sysexit_with_custom_code():
...
+
+def test_resolve_template_dir_absolute():
+    result = util._resolve_template_dir('/foo/bar')
+
+    parts = pytest.helpers.os_split(result)
+
+    assert ('foo', 'bar') == parts[-2:]
+
+
+def test_process_templates():
```
Comment (retr0h) [inline test/unit/test_util.py]: We do have a temp_dir fixture you could use, which would give you a temporary dir to do this work.

File: molecule/util.py [hunk 2]
```diff
@@ -126,6 +129,39 @@ def write_template(src, dest, kwargs={}, _module='molecule', _dir='template'):
...
+
+    :param template_dir: An absolute or relative path to a directory where the templates are located. If the provided
+                         directory is a relative path, it is resolved using a known location.
+    :type template_dir: str
+    :param extra_context: A set of values that are used to override default or user specified values.
+    :type extra_context: dict or None
+    :param output_dir: An absolute path to a directory where the templates should be written to.
+    :type output_dir: str
+    :param overwrite: Whether or not to overwrite existing templates. Defaults to true.
+    :type overwrite: bool
```
Comment (retr0h) [inline molecule/util.py]: I know this is totally OCD, do you mind keeping the length of the comments under 80 chars as well.  Currently the project follows this pattern.

File: test/unit/test_util.py (hunk 1 above)
Comment (signed8bit) [inline test/unit/test_util.py]: Oh cool. Let me move over to using that.

File: molecule/util.py (hunk 2 above)
Comment (signed8bit) [inline molecule/util.py]: Sure. I saw that you were allowing a max line length of 120 in the flake8 config so I was just going off that. I am happy to use 80.

Comment (retr0h) [inline molecule/util.py]: Yeah, 120 in tox.ini is a boo boo 🍼

**Comment by signed8bit** [comment]
syntax: commands succeeded
  py27-ansible19-unit: commands succeeded
  py27-ansible19-functional: commands succeeded
  py27-ansible20-unit: commands succeeded
  py27-ansible20-functional: commands succeeded
  py27-ansible21-unit: commands succeeded
  py27-ansible21-functional: commands succeeded
  doc: commands succeeded
  congratulations :)

**Pull request merged**